ui/shared/components/form/FormWizard.jsx

```diff
@@ -30,24 +32,56 @@ class FormWizard extends React.PureComponent {
 
   resolvedPageSubmit = () => Promise.resolve()
 
+  setSubmitSucceeded = responseJson => this.setState({ responseJson, formSubmitSucceeded: true })
+
   onFormSubmit = (values) => {
-    const { onSubmit } = this.props
-    const { asyncValues } = this.state
-    return onSubmit({ ...asyncValues, ...values })
+    const { onSubmit, onClose } = this.props
+    const { asyncValues, formSubmitSucceeded, responseJson } = this.state
+    if (!formSubmitSucceeded) {
```

Comment:
the change to this function add unneeded complexity and make it incredibly hard to read. There should be no changes to how submission on the create page works. The change here should be conditional on whether or not there is a success message to show, which as I've said only happens on the add data form not the create form. All you need here is:
```
let submitPromise = onSubmit({ ...asyncValues, ...values })
if (successMessage) {
  submitPromise = submitPromise.then(this.setSubmitSucceeded)
}
return submitPromise
```